pkgs/development/compilers/llvm/common/llvm/default.nix

```diff
@@ -329,7 +329,7 @@ stdenv.mkDerivation (finalAttrs: {
   '';
 
   # E.g. Mesa uses the build-id as a cache key (see #93946):
-  LDFLAGS = optionalString (enableSharedLibraries && !stdenv.hostPlatform.isDarwin) "-Wl,--build-id=sha1";
+  LDFLAGS = optionalString (enableSharedLibraries && !stdenv.hostPlatform.isDarwin && !stdenv.buildPlatform.isDarwin) "-Wl,--build-id=sha1";
```

Comment:
Is there no way to specify separate link flags for the build and host platforms?  The comment above sounds like it would be suboptimal to produce builds for Linux hosts that didn't use --build-id.

Reply:
I'm not entirely sure, but I think it might make sense to drop the check for `hostPlatform` here. If you're cross-compiling from Linux, it would be fine to include the build flag (i.e., the compiler is being build to run on Linux, but produce binaries for Darwin).

I think you _should_ be able to compile binaries for macOS from Linux, but I haven't tried this yet.

Reply:
It's not something you can currently do in Nixpkgs.  Checking hostPlatform seems right to me, because it's the host linker that needs to support the argument.  Checking the build linker seems like a workaround for not being able to distinguish between build and host flags, which I bet there's a way to do.